### ISSUE TITLE
bug: congr application type mismatch

### ISSUE THREAD

**Issue by vlad902** [leanprover-community/duper#30]
The following minimal reproducer:
```lean
import Duper

class Magma (α : Type _) where
op : α → α → α
infix:65 " ◇ " => Magma.op

abbrev Equation1689 (G: Type _) [Magma G] := ∀ x y z : G, x = (y ◇ x) ◇ ((x ◇ z) ◇ z)
abbrev Equation2 (G: Type _) [Magma G] := ∀ x y : G, x = y

example (G: Type _) [Magma G] (h: Equation1689 G) : Equation2 G := by duper [h] {portfolioInstance := 2}
```

Hits the following exception:
```
repro.lean:11:71: error: application type mismatch
  Eq.mp (congrArg (fun x => e!_0 x (e!_0 ✝ [anonymous]) = x) (Eq.symm heq)) h
argument
  h
has type
  e!_0 (e!_0 (e!_0 (e!_0 ✝ [anonymous]) (e!_0 ✝ [anonymous])) (e!_0 ✝ [anonymous])) (e!_0 ✝ [anonymous]) =
    e!_0 (e!_0 ✝ [anonymous]) (e!_0 ✝ [anonymous]) : Prop
but is expected to have type
  e!_0 (e!_0 (e!_0 ✝ [anonymous]) (e!_0 ✝ [anonymous])) (e!_0 ✝ [anonymous]) =
    e!_0 (e!_0 ✝ [anonymous]) (e!_0 ✝ [anonymous]) : Prop
```

**Comment by vlad902** [comment]
Note that this example is taken from https://github.com/teorth/equational_theories/

In particular I mention duper [here](https://leanprover.zulipchat.com/#narrow/channel/458659-Equational/topic/Austin.20pairs/near/479643838) on Zulip.

**Comment by JOSHCLUNE** [comment]
This bug is now resolved in the most recent Duper version (v0.0.20)
